mlflow/genai/datasets/evaluation_dataset.py

```diff
@@ -1,118 +1,235 @@
+        if isinstance(dataset, _EntityEvaluationDataset):
+            self._databricks_dataset = None
+            self._mlflow_dataset = dataset
+        else:
+            self._databricks_dataset = dataset
+            self._mlflow_dataset = None
         self._df = None
-        self._digest = None
 
-    @property
-    def dataset_id(self) -> str:
-        """The unique identifier of the dataset."""
-        return self._dataset.dataset_id
+    def __eq__(self, other):
+        """Check equality with another dataset."""
+        if isinstance(other, _EntityEvaluationDataset) and self._mlflow_dataset:
+            return self._mlflow_dataset == other
+        if isinstance(other, EvaluationDataset):
+            if self._mlflow_dataset and other._mlflow_dataset:
+                return self._mlflow_dataset == other._mlflow_dataset
+            if self._databricks_dataset and other._databricks_dataset:
+                return self._databricks_dataset == other._databricks_dataset
+        return False
+
+    def __setattr__(self, name, value):
+        """Allow setting internal attributes on the wrapped dataset."""
+        object.__setattr__(self, name, value)
+        if name == "_records" and hasattr(self, "_mlflow_dataset") and self._mlflow_dataset:
+            self._mlflow_dataset._records = value
+
+    def __getattr__(self, name):
+        """
+        Dynamic attribute delegation for simple pass-through properties.
 
-    @property
-    def digest(self) -> str | None:
-        """String digest (hash) of the dataset provided by the caller that uniquely identifies"""
-        # NB: The managed Dataset entity in Agent SDK doesn't propagate the digest
-        # information. So we compute the digest of the dataframe view.
-        if self._digest is None:
-            self._digest = self._dataset.digest or compute_pandas_digest(self.to_df())
-        return self._digest
+        This handles attributes that don't require special logic and can be
+        directly delegated to the underlying dataset implementation.
+        """
+        if name.startswith("_"):
+            raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
 
-    @property
-    def name(self) -> str | None:
-        """The UC table name of the dataset."""
-        return self._dataset.name
+        if self._mlflow_dataset and hasattr(self._mlflow_dataset, name):
+            return getattr(self._mlflow_dataset, name)
+        elif self._databricks_dataset and hasattr(self._databricks_dataset, name):
+            return getattr(self._databricks_dataset, name)
 
-    @property
-    def schema(self) -> str | None:
-        """The schema of the dataset."""
-        return self._dataset.schema
+        raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
 
     @property
-    def profile(self) -> str | None:
-        """The profile of the dataset, summary statistics."""
-        return self._dataset.profile
```

Comment:
sereral existing `EvaluationDataset` properties / APIs were removed in the PR:

`name`, `schema`, `profile`

is it by intention ?